lib/matobjplist.gi

```diff
@@ -1286,3 +1286,314 @@ InstallMethod( NewCompanionMatrix,
     return ll;
   end );
 
+############################################################################
+# Elementary matrix operations
+############################################################################
+
+############################################################################
+##
+#M  MultMatrixRow( <M>, <row>, <scalar> )
+##
+InstallMethod( MultMatrixRow, "for a mutable IsPlistMatrixRep, one row number, and a scalar",
+  [ IsRowListMatrix and IsMutable, IsInt, IsObject ],
+  function( mat, row, scalar )
+    
+    # Checks
+    if not( 0 < row and row <= NrRows(mat) ) then
+        ErrorNoReturn("the second argument row has to fulfill 0 < row <= NrRows(mat) ");
+    fi;
+  
+    mat[row] := scalar * mat[row];
+    
+  end );
+
+
+############################################################################
+##
+#M  MultMatrixColumn( <M>, <row>, <scalar> )
+##
+InstallMethod( MultMatrixColumn, "for a mutable matrix object, one column number, and a scalar",
+  [ IsRowListMatrix and IsMutable, IsInt, IsObject ],
+  function( mat, column, scalar )
+    local i;
+    
+    # Checks
+    if not( 0 < column and column <= NrCols(mat) ) then
+        ErrorNoReturn("the second argument column has to fulfill 0 < column <= NrCols(mat) ");
+    fi;
+  
+    # TODO ASK MAX
+    
+  end );
+  
+  
+
+############################################################################
+##
+#M  MultMatrixRowLeft( <M>, <row>, <scalar> )
+##
+InstallMethod( MultMatrixRowLeft, "for a mutable matrix object, one row number, and a scalar",
+  [ IsRowListMatrix and IsMutable, IsInt, IsObject ],
+  function( mat, row, scalar )
+ 
+    MultMatrixRow(mat,row,scalar);
+    
+  end );
+
+############################################################################
+##
+#M  MultMatrixRowRight( <M>, <row>, <scalar> )
+##
+InstallMethod( MultMatrixRowRight, "for a mutable matrix object, one row number, and a scalar",
+  [ IsRowListMatrix and IsMutable, IsInt, IsObject ],
+  function( mat, row, scalar )
+      local i;
+    
+    # Checks
+    if not( 0 < row and row <= NrRows(mat) ) then
+        ErrorNoReturn("the second argument row has to fulfill 0 < row <= NrRows(mat) ");
+    fi;
+  
+    mat[row] := mat[row] * scalar;
+    
+  end );
+
+############################################################################
+##
+#M  MultMatrixColumnLeft( <M>, <row>, <scalar> )
+##
+InstallMethod( MultMatrixColumnLeft, "for a mutable matrix object, one column number, and a scalar",
+  [ IsRowListMatrix and IsMutable, IsInt, IsObject ],
+  function( mat, column, scalar )
+    local i;
+    
+    # Checks
+    if not( 0 < column and column <= NrCols(mat) ) then
+        ErrorNoReturn("the second argument column has to fulfill 0 < column <= NrCols(mat) ");
+    fi;
+  
+    for i in [1..NrRows(mat)] do
+        mat[i,column] := scalar * mat[i,column];
+    od;
```

Comment:
This is like in the generic method for `IsMatrixObj`. Thus we need no special method for `IsRowListMatrix` (also in other places).